tests/unit/bootstrap_record_test.cc

```diff
@@ -0,0 +1,269 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+#include "fs/bootstrap_record.hh"
+
+#include <boost/crc.hpp>
+#include <seastar/core/thread.hh>
+#include <seastar/core/units.hh>
+#include <seastar/fs/block_device.hh>
+#include <seastar/fs/temporary_file.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/test_runner.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+using namespace seastar;
+using namespace seastar::fs;
+
+namespace {
+
+constexpr off_t device_size = 1 * MB;
+
+class mock_block_device_impl : public block_device_impl {
+public:
+    using buf_type = basic_sstring<uint8_t, size_t, 32, false>;
+    buf_type buf;
+    ~mock_block_device_impl() override = default;
+
+    future<size_t> write(uint64_t pos, const void* buffer, size_t len, const io_priority_class& pc) override {
+        if (buf.size() < pos + len)
+            buf.resize(pos + len);
+        std::memcpy(buf.data() + pos, buffer, len);
+        return make_ready_future<size_t>(len);
+    }
+
+    future<size_t> read(uint64_t pos, void* buffer, size_t len, const io_priority_class& pc) override {
+        std::memcpy(buffer, buf.data() + pos, len);
+        return make_ready_future<size_t>(len);
+    }
+
+    future<> flush() override {
+        return make_ready_future<>();
+    }
+
+    future<> close() override {
+        return make_ready_future<>();
+    }
+};
+
+future<block_device> prepare_device() {
+    return async([&] {
+        sstring device_path = [&]() -> sstring {
```

Comment:
```suggestion
        sstring device_path = [&] {
```
This is redundant because inferring return type takes place as if it would be declared auto, which results in return type == sstring.